packages/smooth_app/lib/widgets/smooth_product_carousel.dart

```diff
@@ -307,6 +291,24 @@ class _SearchCardTagLine extends StatelessWidget {
       ),
     );
   }
+
+  /// We fetch first if the app is deprecated, then try to get the tagline
+  /// Will return [false] if the app is deprecated or a [TagLineItem]
+  Future<dynamic> _fetchData() async {
```

Comment:
```suggestion
  Future<TagLineItem?> _fetchData() async {
```

`dynamic` is ugly; please return `null` if deprecated. I may be wrong but that shouldn't be a problem.